pkg/build/api/types.go

```diff
@@ -178,6 +188,26 @@ type GitHubWebHookCause struct {
 	Secret string
 }
 
+// GitLabWebHookCause has information about a GitLab webhook that triggered a
+// build.
+type GitLabWebHookCause struct {
+	// Revision is the git source revision information of the trigger.
+	Revision *SourceRevision
+
+	// Secret is the obfuscated webhook secret that triggered a build.
+	Secret string
+}
+
+// BitbucketWebHookCause has information about a Bitbucket webhook that triggered a
+// build.
+type BitbucketWebHookCause struct {
+	// Revision is the git source revision information of the trigger.
+	Revision *SourceRevision
+
+	// Secret is the obfuscated webhook secret that triggered a build.
+	Secret string
+}
```

Comment:
Based on a side conversation I had with @bparees yesterday, changing `GitHubWebHookCause` and `GenericWebHookCause` to leverage `CommonWebHookCause` is still dependent @smarterclayton confirming to him where to land on the protobuf / etcd / storage concern.  And that is the only change still on the table.